### ISSUE TITLE
mdDialog generates unhelpful aria-label

### ISSUE THREAD

**Issue by calebegg** [angular/material#10582]
**Actual Behavior**:
 -  `What is the issue? *`

By default, `$mdDialog` generates an `aria-label` for confirm/alert dialogs by shortening the title. For JAWS at least, this is counterproductive because JAWS reads something like "This is an dot dot dot this is an alert title" instead of just "This is an alert title".

There's also no way to suppress this; if I do `.ariaLabel('')`, it still generates the label.

 -  `What is the expected behavior?`

If I don't specify an aria-label, I just want the screen reader to read the contents of the dialog.

**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:`

http://codepen.io/anon/pen/jmNZpG

**Comment by EladBezalel** [comment]
@calebegg would you be interested in doing a pr for this one?

**Comment by alopezsanchez** [comment]
Hello @calebegg, @EladBezalel. I'm working on this issue.

Do you mean not to generate the `aria-label` attribute if you don't specify one? Or generate it with only the title of the dialog (if exists)?

**Comment by ThomasBurleson** [comment]
@EladBezalel - ping

**Comment by EladBezalel** [comment]
@alopezsanchez i think the latter, generate with title only if exists or if you specify the aria label

**Comment by alopezsanchez** [comment]
@EladBezalel @ThomasBurleson, PR done.

**Comment by calebegg** [comment]
Sorry for the delay, I was on vacation.

We can't figure out why we'd ever want an `aria-label`, let alone one that simply restates the contents of the dialog (either title or body). This just adds noise for screen reader users who now have to listen to the title or body twice. The issue was reported to us by a screen reader user, and when I went to fix it, there didn't seem to be a way to suppress the creation of this `aria-label`.

I'm not an accessibility expert, so maybe there's some reason for it that I don't see, but to resolve our issue, I'd like, at the very least, a way to suppress the generation of an `aria-label` completely. Perhaps by passing an empty string to `.ariaLabel('')`.

**Comment by alopezsanchez** [comment]
I understand what you say, but `aria-label` is read too when the dialog is closed... So I don't know if we should avoit it (I'm not an accessibility expert neither).

What do you think @EladBezalel, @ThomasBurleson?

**Comment by EladBezalel** [comment]
@marcysutton mind jumping in?

**Comment by marcysutton** [comment]
I believe the `aria-label` was added to ensure the dialog was exposed with an accessible name, in case there wasn't a heading with an ID to use with `aria-labelledby`. I agree that the truncation makes it less useful. Ideally, a heading in the dialog could be given an id and linked with `aria-labelledby` on the dialog to expose its name (making it more useful as a screen reader landmark). 

To be honest, it's been so long I don't quite remember why it was coded in that exact way–but I'm guessing the `aria-label` method was intended to be a failsafe in case there was no header-id-association.

**Comment by calebegg** [comment]
This solution does not resolve our issue, as I described it in [my comment above](https://github.com/angular/material/issues/10582#issuecomment-308805140). We need a way to have no ariaLabel. Users are complaining that the dialog title is read twice.
